client/src/components/Navigation.tsx

```diff
@@ -125,135 +78,47 @@ export function Navigation({ categories }: Props) {
               LUMINELLE
             </h1>
           )}
-          <ul
-            className={`flex ${
-              isHomePage ? 'justify-center' : ''
-            } mt-4 md:mt-0`}>
-            {categories.map((category) => (
-              <li
-                key={category.name}
-                onMouseOver={() => handleMouseOver(category.name)}
-                onMouseLeave={handleMouseLeave}
-                className="cursor-pointer px-4 py-2">
-                {category.name}
-                {activeCategory === category.name && (
-                  <div className="absolute left-0 right-0 top-10 z-10 bg-white w-full border-t-2 border-slate-300 pt-1">
-                    {category.subcategories.map((subcategory) => (
-                      <Link
-                        key={subcategory.name}
-                        to={subcategory.path}
-                        className="block px-4 py-2 focus:underline">
-                        {subcategory.name}
-                      </Link>
-                    ))}
-                  </div>
-                )}
-              </li>
-            ))}
-          </ul>
+          <CategoryMenu
+            categories={categories}
+            activeCategory={activeCategory}
+            handleMouseOver={handleMouseOver}
+            handleMouseLeave={handleMouseLeave}
+          />
           <div className="flex">
-            <FaRegUser
-              className="cursor-pointer mx-2"
-              onClick={toggleUserMenu}
-            />
+            {user ? (
+              <FaUser
+                className="cursor-pointer mx-2"
+                onClick={toggleUserMenu}
+              />
+            ) : (
+              <FaRegUser
+                className="cursor-pointer mx-2"
+                onClick={toggleUserMenu}
+              />
+            )}
             <IoSearch className="cursor-pointer mx-2" onClick={toggleSearch} />
-            <FaRegHeart className="cursor-pointer mx-2" />
+            {wishlist.length === 0 ? (
+              <FaRegHeart
+                className="cursor-pointer mx-2"
+                onClick={() => navigate('/wishlist')}
+              />
+            ) : (
+              <FaHeart
+                className="cursor-pointer mx-2"
+                onClick={() => navigate('/wishlist')}
+              />
+            )}
             <IoBagOutline className="cursor-pointer mx-2" />
           </div>
-          {isUserOpen && (
```

Comment:
Oh! You moved this into a separate component before seeing my comment in the prior PR! Well done!